realm/realm-library/src/objectServer/java/io/realm/internal/SyncObjectServerFacade.java

```diff
@@ -86,11 +87,17 @@ public void realmClosed(RealmConfiguration configuration) {
     public Object[] getUserAndServerUrl(RealmConfiguration config) {
         if (config instanceof SyncConfiguration) {
             SyncConfiguration syncConfig = (SyncConfiguration) config;
-            String rosServerUrl = syncConfig.getServerUrl().toString();
-            String rosUserIdentity = syncConfig.getUser().getIdentity();
-            String syncRealmAuthUrl = syncConfig.getUser().getAuthenticationUrl().toString();
-            String rosRefreshToken = syncConfig.getUser().getAccessToken().value();
-            return new Object[]{rosUserIdentity, rosServerUrl, syncRealmAuthUrl, rosRefreshToken, syncConfig.syncClientValidateSsl(), syncConfig.getServerCertificateFilePath()};
+            // make sure the user is still valid
+            SyncUser user = syncConfig.getUser();
+            if (!user.isValid()) {
+                throw new IllegalStateException("SyncUser is no longer valid, probably logged out.");
```

Comment:
the issue is when the `refresh_token` is not usable the user need to login again, this is different from the use case where the `access_token` expires & we can wait for the client to refresh it.

btw this will throw only when creating the first instance in the current thread, when returning a "thread cached" instance we don't do any check. 

I can add a warning/error log instead if you feel strongly against.